src/pybind/mgr/progress/module.py

```diff
@@ -364,7 +364,8 @@ def _osd_in_out(self, old_map, old_dump, new_map, osd_id, marked):
         # In the case of the osd coming back in, we might need to cancel 
         # previous recovery event for that osd
         if marked == "in":
-            for ev_id, ev in self._events.items():
+            for ev_id in list(self._events):
+                ev = self._events[ev_id]
```

Comment:
is it possible that an event gets removed from `self._events` by another thread while we are iterating the list of key?

Reply:
I think the reason I made a fix for this is because of the case when you are iterating the list of events and found an event that can be completed. So while the program is still iterating through the dictionary, the size of the dictionary is changed giving it a runtime error.

As for why it gets removed by another thread, I don't think this ever happened when I was running them, but I could be wrong